xbmc/video/VideoInfoScanner.cpp

```diff
@@ -437,6 +452,44 @@ namespace VIDEO
     return !m_bStop;
   }
 
+  std::map<std::string, std::string> CVideoInfoScanner::UpdateSetFromLocalArtwork(
+      const CVideoInfoTag& setDetails)
+  {
+    CGUIListItem::ArtMap movieSetArt;
+    m_database.Open();
+    if (IsArtNeededAndLocalArtwork(setDetails))
+    {
+      // directory exists in Movie Set Information Folder and artwork currently blank
+      const std::string movieSetInfoPath = GetMovieSetInfoFolder(setDetails.m_strTitle);
+      const std::vector<std::string> movieSetArtTypes =
+          CVideoThumbLoader::GetArtTypes(MediaTypeVideoCollection);
+      AddLocalItemArtwork(movieSetArt, movieSetArtTypes, movieSetInfoPath, true, false);
+      m_database.SetArtForItem(setDetails.m_iDbId, MediaTypeVideoCollection, movieSetArt);
+    }
+    m_database.Close();
+    return movieSetArt;
+  }
+
+  bool CVideoInfoScanner::IsArtNeededAndLocalArtwork(const CVideoInfoTag& setDetails)
+  {
+    bool result = false;
+    if (!m_database.Open())
+      return result;
+    std::map<std::string, std::string> movieArt, setArt;
+    const bool isArt =
+        m_database.GetArtForItem(setDetails.m_iDbId, MediaTypeVideoCollection, setArt);
+    if (!isArt || !setArt.count("poster"))
+    {
+      // Movie set currently has no artwork poster (other art types do not seem to be used)
```

Comment:
Estuary uses the fanart when defined in addition to poster. I don't know about the other skins or about the automatic download.